gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -140,21 +138,26 @@ public void testByRoutingRulesEngineFileChange()
         assertEquals("etl",
                 routingGroupSelector.findRoutingGroup(mockRequest));
 
-        Thread.sleep(1);
-        // java.nio.file.attribute.FileTime offers a maximum precision of 1 ms. If the first
-        // half of this test runs in <1ms, the gateway may not recognize that the file
-        // has changed.
+        long etl2LastModified = etlLastModified;
+        while (!(etl2LastModified > etlLastModified)) {
+            Thread.sleep(1);
+            // RuleReloadingRoutingGroupSelector compares file timestamps to see if the rules file
+            // has been updated, and java.nio.file.attribute.FileTime offers a maximum precision of 1 ms.
+            // If the first half of this test runs in <1ms, the gateway may not recognize that the file
+            // has changed.
+
+            fw = new FileWriter(file, UTF_8);
+            fw.write(
+                    "---\n"
+                            + "name: \"airflow2\"\n"
+                            + "description: \"updated rule\"\n"
+                            + "condition: \"request.getHeader(\\\"X-Trino-Source\\\") == \\\"airflow\\\"\"\n"
+                            + "actions:\n"
+                            + "  - \"result.put(\\\"routingGroup\\\", \\\"etl2\\\")\""); // change from etl to etl2
+            fw.close();
+            etl2LastModified = file.lastModified();
+        }
 
-        fw = new FileWriter(file, UTF_8);
-        fw.write(
-                "---\n"
-                        + "name: \"airflow2\"\n"
-                        + "description: \"updated rule\"\n"
-                        + "condition: \"request.getHeader(\\\"X-Trino-Source\\\") == \\\"airflow\\\"\"\n"
-                        + "actions:\n"
-                        + "  - \"result.put(\\\"routingGroup\\\", \\\"etl2\\\")\""); // change from etl to etl2
-        fw.close();
-        long etl2LastModified = file.lastModified();
         log.info("etl2 rule file created at: " + etl2LastModified);
         assertNotEquals(etl2LastModified, etlLastModified);
```

Comment:
Also .. not that seems pointless .. we are basically forcing it to be different .. and then we asserting that it is .. at that stage the test might be useless altogether.